csrc/kernel_cache.cpp

```diff
@@ -350,8 +364,33 @@ FusionKernelRuntime* FusionExecutorCache::getKernelRuntimeFor(
     }
   }
 
-  // Access kernels associated with the common device id
-  auto& kernel_runtimes = kernel_runtimes_[args.getDeviceIndex()];
+  // Compute concretization info given inputs. This object points to Vals in
+  // the unconcretized Fusion, so we will not use it directly, but rather it
+  // will be used only as a cache key.
+  std::optional<DynamicTransformConcretizationInfo> conc_info = std::nullopt;
+  size_t conc_info_index = 0;
+  if (has_dynamic_reshape_) {
+    conc_info = DynamicTransform::getConcretizationInfo(fusion_.get(), &args);
```

Comment:
I think this is fine for now as it's only called when there's dynamic reshape, but the latency of `getConcretizationInfo` may be an issue as it traverses the whole fusion every time with a newly created ExpressionEvaluator. It also creates `ExactRootDomainMap` as `ExpressionEvalutor::propagateBoundValuesThroughExactMaps` is called. We should investigate the latency and think about caching of some of the analyses that don't depend on actual inputs. No need to do in this PR, but please create an issue to keep track of it.

Reply:
#244